src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -750,5 +733,328 @@ private static string ToHtmlCase(string name)
+                    }
+                    else
+                    {
+                        descriptor = ParsePlainSelector(errorSink);
+                    }
+
+                    if (descriptor == null)
+                    {
+                        // Failed to create the descriptor due to an invalid required attribute.
+                        return false;
+                    }
+                    else
+                    {
+                        descriptors.Add(descriptor);
+                    }
+
+                    PassOptionalWhitespace();
+
+                    if (At(','))
+                    {
+                        Next();
+                        PassOptionalWhitespace();
+                    }
+                    else if (!AtEnd)
+                    {
+                        errorSink.OnError(SourceLocation.Zero, $"TODO: Unknown character '{Current}'. Separate required attributes with commas.", 0);
+                        return false;
+                    }
+                }
+
+                requiredAttributeDescriptors = descriptors;
+                return true;
+            }
+
+            private TagHelperRequiredAttributeDescriptor ParsePlainSelector(ErrorSink errorSink)
+            {
+                var nextSeparator = _requiredAttributes.IndexOf(',', _index);
+                string attributeName;
+
+                if (nextSeparator == -1)
+                {
+                    attributeName = _requiredAttributes.Substring(_index).Trim();
+                    _index = _requiredAttributes.Length;
+                }
+                else
+                {
+                    attributeName = _requiredAttributes.Substring(_index, nextSeparator - _index).Trim();
+                    _index = nextSeparator;
+                }
+
+                var hasWildcard = false;
+                if (attributeName[attributeName.Length - 1] == RequiredAttributeWildcardSuffix)
+                {
+                    attributeName = attributeName.Substring(0, attributeName.Length - 1);
+                    hasWildcard = true;
+                }
+
+                TagHelperRequiredAttributeDescriptor descriptor = null;
+                if (ValidateName(attributeName, targetingAttributes: true, errorSink: errorSink))
+                {
+                    descriptor = new TagHelperRequiredAttributeDescriptor
+                    {
+                        Name = attributeName
+                    };
+
+                    if (hasWildcard)
+                    {
+                        descriptor.Operator = RequiredAttributeWildcardSuffix;
+                    }
+                }
+
+                return descriptor;
+            }
+
+            private string ParseCSSAttributeName(ErrorSink errorSink)
+            {
+                var nameStartIndex = _index;
+                while (!AtEnd && !char.IsWhiteSpace(Current) && !At('=') && !At(']'))
```

Comment:
`IsWhiteSpace()` forces use of the `while` instead of the much more straightforward `string` methods such as `IsIndexOfAny()`. Also end up going backward and forward in the string.